### PR TITLE
feat: Add project and project relations entities

### PR DESCRIPTION
**Pull request by valya**
## Summary
Adds the entities, repositories, and migrations for setting up Projects.

## Review / Merge checklist
- [x] PR title and summary are descriptive. **Remember, the title automatically goes into the changelog. Use `(no-changelog)` otherwise.** ([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))

### CODE REVIEW

**Review by krynble — APPROVED** [review]
Looks good

File: packages/cli/src/databases/migrations/common/1705928727784-CreateProject.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+export class CreateProject1705928727784 implements ReversibleMigration {
+	async up({ dbType, schemaBuilder: { createTable, column } }: MigrationContext) {
+		await createTable(tableName).withColumns(
+			column('id').varchar(36).primary.notNull,
+			column('name').varchar(255),
+			column('type').varchar(36),
+		).withTimestamps;
+
+		let userIdColumn = column('userId').varchar().primary.notNull;
```
Comment (netroy) [inline packages/cli/src/databases/migrations/common/1705928727784-CreateProject.ts:14]: could we use `uuid` for all DBs, and handle this in the DSL instead?

Comment (valya) [reply packages/cli/src/databases/migrations/common/1705928727784-CreateProject.ts:14]: The problem is `uuid` is an actual column type in SQLite as well (though I think it's basically a string under the hood?), it's just that the user `id` column is just different between the DBs. It's a `UUID` on postgres, a `varchar(36)` on mysql, and a `varchar` with no size limit on SQLite.

Comment (netroy) [reply packages/cli/src/databases/migrations/common/1705928727784-CreateProject.ts:14]: I'd still prefer us to handle this in the DSL, and not add additional complexity in the migrations themselves.
I propose that we make [this change](https://github.com/n8n-io/n8n/commit/20f43ca8094358da6d7ed76eab88e2f2d21f5f3d) instead.

**Review by netroy — CHANGES_REQUESTED** [review]

**Review by netroy — APPROVED** [review]

**Pull request merged**